core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorPageSourceProvider.java

```diff
@@ -17,6 +17,14 @@
 
 public interface ConnectorPageSourceProvider
 {
+    /**
+     * @return If this page source provider is stateful per task execution this method should return a newly initialized stateful copy of the provider
+     */
+    default ConnectorPageSourceProvider getStatefulInstance()
```

Comment:
does the stateful instance need to be closeable?

Reply:
> per task execution

if this is created for a table scan, the method should take a table handle object.

Reply:
This is a leftover, I will delete it.

Reply:
Deleted the unused method and cleaned up the commits to match